Upgrade demo site's `styled-components` NPM package to v6

The `/demo/frontend` uses v5 of the `styled-components` npm, and needs to be updated.

See also https://styled-components.com/docs/faqs#what-do-i-need-to-do-to-migrate-to-v6

Updating package.json styled components to 6.1.1 produces this error:

```
cd demo/frontend
just up-build
...
0.471 yarn run v1.22.19
0.493 $ tsc && vite build
5.740 src/Components/Map/Filters/AvgTime.ts(10,15): error TS2339: Property 'isEnabled' does not exist on type 'ExecutionContext & FastOmit<DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>, never>'.
5.740 src/Components/Map/Filters/Filters.tsx(88,13): error TS2769: No overload matches this call.
5.740   Overload 1 of 2, '(props: PolymorphicComponentProps<"web", FastOmit<DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>, never>, void, void, {}, {}>): Element', gave the following error.
5.740     Type '{ children: string; isEnabled: boolean; onClick: () => void; }' is not assignable to type 'IntrinsicAttributes & FastOmit<Substitute<FastOmit<DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>, never>, FastOmit<...>>, keyof ExecutionProps> & FastOmit<...> & { ...; }'.
5.740       Property 'isEnabled' does not exist on type 'IntrinsicAttributes & FastOmit<Substitute<FastOmit<DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>, never>, FastOmit<...>>, keyof ExecutionProps> & FastOmit<...> & { ...; }'.
5.740   Overload 2 of 2, '(props: FastOmit<DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>, never>): ReactNode', gave the following error.
5.740     Type '{ children: string; isEnabled: boolean; onClick: () => void; }' is not assignable to type 'IntrinsicAttributes & FastOmit<DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>, never>'.
5.740       Property 'isEnabled' does not exist on type 'IntrinsicAttributes & FastOmit<DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>, never>'.
5.740 src/Components/Map/Filters/Layers/Layer.ts(8,15): error TS2339: Property 'isLayerVisible' does not exist on type 'ExecutionContext & FastOmit<DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>, never>'.
5.740 src/Components/Map/Filters/Layers/Legend.ts(10,12): error TS2339: Property 'fromColor' does not exist on type 'ExecutionContext & FastOmit<DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>, never>'.
5.740 src/Components/Map/Filters/Layers/Legend.ts(10,23): error TS2339: Property 'toColor' does not exist on type 'ExecutionContext & FastOmit<DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement>, never>'.
5.782 info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
5.782 error Command failed with exit code 2.
```

--- a/demo/frontend/package.json
+++ b/demo/frontend/package.json
@@ -21,7 +21,7 @@
     "react-dom": "18.2.0",
     "react-map-gl": "^7.1.6",
     "react-scroll-parallax": "3.4.3",
-    "styled-components": "5.3.11"
+    "styled-components": "6.1.1"
   },
   "devDependencies": {
     "@types/node": "^20.9.0",

--- a/demo/frontend/src/Components/Map/Filters/AvgTime.ts
+++ b/demo/frontend/src/Components/Map/Filters/AvgTime.ts
@@ -1,6 +1,6 @@
 import styled from 'styled-components';
 
-export default styled.div`
+export default styled.div<{ isEnabled: boolean }>`
   padding: 5px;
   border-radius: 5px;
 
@@ -8,7 +8,6 @@ export default styled.div`
   font-weight: bold;
 
   color: ${({ isEnabled }) => (isEnabled ? '#DADFEE' : '#6C7495')};
-  
   cursor: pointer;
 
   &:hover {

--- a/demo/frontend/src/Components/Map/Filters/Layers/Layer.ts
+++ b/demo/frontend/src/Components/Map/Filters/Layers/Layer.ts
@@ -1,12 +1,11 @@
 import styled from 'styled-components';
 
-export default styled.div`
+export default styled.div<{ isLayerVisible: boolean }>`
   padding: 5px;
   border-radius: 5px;
   margin-bottom: 15px;
 
   color: ${({ isLayerVisible }) => (isLayerVisible ? '#DADFEE' : '#6C7495')};
-  
   cursor: pointer;
 
   &:hover {

--- a/demo/frontend/src/Components/Map/Filters/Layers/Legend.ts
+++ b/demo/frontend/src/Components/Map/Filters/Layers/Legend.ts
@@ -1,6 +1,6 @@
 import styled from 'styled-components';
 
-export default styled.div`
+export default styled.div<{ fromColor: string; toColor: string }>`
   height: 4px;
   border-radius: 5px;
   margin-top: 6px;

